### PR TITLE
Use MapperTestCase for cartesian mappers tests

### PR DESCRIPTION
**Pull request by iverase**
This commit changes the cartesian mapper tests (point and shape fields) so that they use `MapperTestCase`. In addition the tests are moved out of InternalclusterTest as they do not need to spin up a cluster any more.

### CODE REVIEW

File: x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapperTests.java [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        });
+    }
+
+    public void testValuesStored() throws Exception {
+        DocumentMapper mapper = createDocumentMapper(fieldMapping(b -> {
+            b.field("type", "point");
+            b.field("store", true);
+        }));
+
+        SourceToParse sourceToParse = new SourceToParse("test", "1",
```
Comment (romseygeek) [inline x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapperTests.java:53]: You can also use the `source` method, which simplifies this a bit:
```
SourceToParse sourceToParser = source(b -> {
    b.startObject(FIELD_NAME).field("x", 2000.1).field("y", 305.6).endObject();
});
```

**Review by romseygeek — APPROVED** [review]
LGTM! Good to see more of these being converted, thanks @iverase

File: x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapperTests.java (hunk 1 above)
Comment (iverase) [reply x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapperTests.java:53]: I am changing it, I was unhappy with that.

**Comment by iverase** [comment]
@elasticmachine update branch

**Comment by iverase** [comment]
@elasticmachin run elasticsearch-ci/bwc

**Comment by iverase** [comment]
@elasticmachine update branch

**Pull request merged**